en/V2-User_Space_Application_Requirements.md

```diff
@@ -20,9 +20,13 @@ Many controls in this chapter are implemented through cryptography. Therefore ad
 | **2.1.2** | Verify that all connected devices within the IoT ecosystem can be uniquely identified including connected to the cloud, hubs, as well as to other devices (sensors). | ✓ | ✓ | ✓ |
 | **2.1.3** | Verify strong user and device authentication is enforced across the IoT ecosystem. | ✓ | ✓ | ✓ |
 | **2.1.4** | Verify that user, service, and device authentication schemes share a common framework centrally managed in the IoT ecosystem. | ✓ | ✓ | ✓ |
-| **2.1.5** | Verify certificate based authentication is preferred over password based authentication within the IoT ecosystem. | ✓ | ✓ | ✓ |
-| **2.1.6** | Verify good password policies are enforced throughout the IoT ecosystem by disallowing hardcoded passwords and duplicate identities or passwords across devices. | ✓ | ✓ | ✓ |
-
+| **2.1.5** | Verify that passwords used for user authentication are at least 12 characters long. | ✓ | ✓ | ✓ |
+| **2.1.6** | Verify that passwords used for user authentication can be changed by the user and that the password change functionality requires the user's current and new password. | ✓ | ✓ | ✓ |
+| **2.1.7** | Verify that passwords used for device authentication are sufficiently long and complex. | ✓ | ✓ | ✓ |
+| **2.1.8** | Verify that credentials used for user or device authentication can be changed. | ✓ | ✓ | ✓ |
+| **2.1.9** | Verify that no hardcoded credentials are used for user for device authentication. | ✓ | ✓ | ✓ |
+| **2.1.10** | Verify that provisioned credentials for device authentication are unique per device. | ✓ | ✓ | ✓ |
+| **2.1.11** | Verify that credentials of compromised or decommissioned devices are revoked. | ✓ | ✓ | ✓ |
```

Comment:
I like this one although I'm not sure how this would be measurable as is. Management applications will need to be designed with the ability to revoke credentials and maintain asset tracking via their secure identity (only trustworthy source if it has a RoT). Assessors might audit or test admin features that enable revocation of credentials. 
The flip side of this control is we may be giving manufacturers a lot of power depending on the usage. For enterprise IoT, this makes sense 100%. Other industry verticals like consumer IoT may have conflicts and could be overextending their reach. Although, I suppose Tesla sort of has this feature already and is considered in the consumer realm. A feature like this could also be an entry point and allow for ransomware like capabilities. 🤔 
Noting that we should consider modifying the level to three. Designing a resilient credential revocation system is not an easy task at scale. Revocation lists must maintain integrity to ensure they are tamper resistant and requires additional logging controls.  

If we choose to keep, how about the following at level 3?
`Verify that authentication schemes are designed to revoke credentials of compromised or decommissioned devices.`